src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -127,9 +127,54 @@ namespace winrt::TerminalApp::implementation
         _tabView.TabItemsChanged({ this, &TerminalPage::_OnTabItemsChanged });
 
         _CreateNewTabFlyout();
-        _OpenNewTab(nullptr);
 
         _tabContent.SizeChanged({ this, &TerminalPage::_OnContentSizeChanged });
+
+        // Actually start the terminal.
+        if (_appArgs.GetStartupActions().size() == 0)
+        {
+            _OpenNewTab(nullptr);
+        }
+        else
+        {
+            _appArgs.ValidateStartupCommands();
+
+            // This will kick off a chain of events to perform each startup
+            // action. As each startup action is completed, the next will be
+            // fired.
+            _processNextStartupAction();
+        }
+    }
+
+    // Method Description:
+    // - Process the next startup action in our list of startup actions. When
+    //   that action is complete, fire the next (if there are any more).
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TerminalPage::_processNextStartupAction()
+    {
+        // If there are no actions left, do nothing.
+        if (_appArgs.GetStartupActions().size() == 0)
+        {
+            return;
+        }
+
+        // Get the next action to be processed
+        auto nextAction = _appArgs.GetStartupActions().front();
```

Comment:
somehow this does work, though you're right that it shouldn't...

Maybe this is c++winrt magic. I'm getting a ref to a winrt object, which then I'm using to create a `winrt::TerminalApp::ActionAndArgs`, which increments the object's refcount, and then the pop_front is decrementing the refcount, leaving the last ref in the local `nextAction`. 

Or at least that's what I'd think is happening here. @DHowett-MSFT confirm?

Reply:
Actually, no, here's a fun thing.

```
int &x() {
}

auto f = x();
```

f is deduced to be `int`, not `int&` because `auto` follows the [template type deduction rules](https://en.cppreference.com/w/cpp/language/template_argument_deduction)

Reply:
so: you're not pulling a reference out of the vector

Reply:
okay thanks for the clarification!

Reply:
Phew ok.